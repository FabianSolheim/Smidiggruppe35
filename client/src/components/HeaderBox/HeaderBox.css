.TopHeader {
    width: 100%;
    height: 764px;
}
.HeaderBox {
    position: relative;
    height: 30%;
    width: 50%;
    min-width: 850px;
    max-width: 850px;
    top: 50%;
    padding: 50px;
    margin: auto;
    transform: translateY(-50%);
    background-color: rgba(255, 255, 255, 0.4);
    backdrop-filter: blur(6px);
}
.BoxHeadline {
    font-size: 34px;
}
.BoxHeadline em {
    color: #034F7A;
    font-style: normal;
}
.BoxText em {
    color: #034F7A;
    font-style: normal;
}
.BoxText {
    font-size: 26px;
    margin-top: 50px;
}
@media only screen and (max-width: 900px) {

    .TopHeader {
        height: 289px;
    }
    .HeaderBox {
        min-width: 350px;
        height: 45%;
        padding: 10px;
    }
    .BoxHeadline {
        font-size: 15px;
    }
    .BoxText {
        font-size: 16px;
        margin-top: 10px;
    }
}

@media only screen and (max-width: 400px) {
    .TopHeader {
        height: 289px;
    }
    .HeaderBox {
        min-width: 260px;
        height: 35%;
        padding: 10px;
    }
    .BoxHeadline {
        font-size: 12px;
    }
    .BoxText {
        font-size: 10px;
        margin-top: 10px;
    }
}