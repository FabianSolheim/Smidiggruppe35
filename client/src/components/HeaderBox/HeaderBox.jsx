import React from "react";
import "./HeaderBox.css";

export function HeaderBox(
  backgroundImageUrl,
  headline,
  headlineColor,
  text,
  textColor
) {
  return (
    <div
      className="TopHeader"
      style={{
        backgroundImage: `url(${backgroundImageUrl})`,
        backgroundRepeat: "no-repeat",
        backgroundSize: "cover",
        backgroundPosition: "center",
        textAlign: "center",
      }}
    >
      <div className="HeaderBox">
        <div
          className="BoxHeadline"
          style={{
            color: headlineColor,
          }}
        >
          {headline}
        </div>
        <div
          className="BoxText"
          style={{
            color: textColor,
          }}
        >
          {text}
        </div>
      </div>
    </div>
  );
}
