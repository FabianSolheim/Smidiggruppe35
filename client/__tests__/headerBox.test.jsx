import React from "react";
import ReactDOM from "react-dom";
import { HeaderBox } from "../src/components/HeaderBox/HeaderBox";

describe("headerBox test", () => {
  it("shows headerBox", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(<div>{HeaderBox}</div>, domElement);
    expect(domElement.innerHTML).toMatchSnapshot();
  });
});
