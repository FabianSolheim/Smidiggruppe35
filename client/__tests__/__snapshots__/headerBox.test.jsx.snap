// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`headerBox test shows headerBox 1`] = `"<div></div>"`;
